cogfiles/copypastas.py

```diff
@@ -1,22 +1,49 @@
-import discord, json
+import discord
+import json
 
+from discord import app_commands
 from discord.ext import commands
+from ioutils import read_json, write_json
+from typing import Literal
 
 
 class Copypastas(commands.Cog, name="Message Copypastas"):
     """Send a copypasta whenever a key phrase is found in a message."""
     
     def __init__(self, bot: commands.Bot):
         self.bot = bot
+        self.is_copypasta_enabled: dict[int, bool] = {} # Whether or not copypastas are enabled for each server
         with open("copypastas.json", "r") as file:
             self.copypastas = json.load(file)
     
+    @commands.Cog.listener()
+    async def on_ready(self):
+        for guild in self.bot.guilds:
+            self.is_copypasta_enabled[guild.id] = read_json(guild.id, "copypasta")
+
     @commands.Cog.listener()
     async def on_message(self, message: discord.Message):
         """Detect key phrases in messages."""
         # The bot shouldn't respond to its own copypastas
         if message.author == self.bot.user:
             return
+        if not self.is_copypasta_enabled[message.guild.id]:
+            return
         for phrase in self.copypastas:
             if phrase in message.content.lower():
-                await message.channel.send(self.copypastas[phrase])
\ No newline at end of file
+                await message.channel.send(self.copypastas[phrase])
+
+    @app_commands.command()
+    @app_commands.checks.has_permissions(manage_guild=True)
+    async def copypastamode(self, interaction: discord.Interaction, mode: Literal["On", "Off"]):

```

Comment:
Looks good, just curious if a boolean parameter would work here as well as opposed to the Literal "On" or "Off" — also, if you do end up changing that, I might suggest renaming the command and parameter to "toggle" instead of "mode" (something like `togglecopypastas`)

Reply:
Yeah boolean works fine as a parameter